cmd/lncli/commands.go

```diff
@@ -538,6 +569,190 @@ func showRoutingTable(ctx *cli.Context) error {
 		return err
 	}
 
-	printRespJson(resp)
+	// TODO(mkl): maybe it is better to print output directly omitting
+	// conversion to RoutingTable. This part is not performance critical so
+	// I think it is ok because it enables code reuse
+	r := rt.NewRoutingTable()
+	for _, channel := range resp.Channels {
+		r.AddChannel(
+			graph.NewID(channel.Id1),
+			graph.NewID(channel.Id2),
+			graph.NewEdgeID(channel.EdgeID),
+			&rt.ChannelInfo{channel.Capacity, channel.Weight},
+		)
+	}
+
+	req2 := &lnrpc.GetInfoRequest{}
+	resp2, err := client.GetInfo(ctxb, req2)
+	if err != nil {
+		return err
+	}
+	self, _ := hex.DecodeString(resp2.LightningId)
+
+	typ := ctx.String("type")
+	viz := ctx.String("viz")
+	if typ != "" || viz != "" {
+		TempFile, err  := ioutil.TempFile("", "") 
```

Comment:
A temporary file is needed because for rendering external program is used which needs existing file for input.  neato (one of graphviz layout programs) is used.